cmd/sops/main.go

```diff
@@ -687,27 +695,36 @@ func keyGroups(c *cli.Context, file string) ([]sops.KeyGroup, error) {
 	return []sops.KeyGroup{group}, nil
 }
 
-func shamirThreshold(c *cli.Context, file string) (int, error) {
-	if c.Int("shamir-secret-sharing-threshold") != 0 {
-		return c.Int("shamir-secret-sharing-threshold"), nil
-	}
+func loadConfig(c *cli.Context, file string, kmsEncryptionContext map[string]*string) (*config.Config, error) {
 	var err error
 	var configPath string
 	if c.String("config") != "" {
 		configPath = c.String("config")
 	} else {
 		configPath, err = config.FindConfigFile(".")
 		if err != nil {
-			// If shamir flag isn't set and we can't find a config file,
-			// assume we don't want Shamir
-			return 0, nil
+			// If we can't find a config file, but we were not request it, assume it does not exist
+			return nil, nil
 		}
 	}
-	conf, err := config.LoadForFile(configPath, file, nil)
+	conf, err := config.LoadForFile(configPath, file, kmsEncryptionContext)
 	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func shamirThreshold(c *cli.Context, file string) (int, error) {
+	if c.Int("shamir-secret-sharing-threshold") != 0 {
+		return c.Int("shamir-secret-sharing-threshold"), nil
+	}
+	conf, err := loadConfig(c, file, nil)
+	if conf == nil {
+		// If shamir flag isn't set and we can't find a config file,
+		// assume we don't want Shamir
 		return 0, err
```

Comment:
Why does this return an error?

Reply:
This encompass two cases which were separated previously.
1. No config was provided. Err will be nil and ShamirThreshold will be the default value of 0.
2. We did find a config file, but failed to load it. In that case the calling function will print the error and exit.

Reply:
OK, that makes sense. Could you document `loadConfig` so that this behavior is clear to other people?

Reply:
Done. Also found bug :(.